rotkehlchen/chain/evm/transactions.py

```diff
@@ -447,6 +460,71 @@ def _get_erc20_transfers_for_ranges(
                 queried_ranges=[(start_ts, end_ts)],
             )
 
+    def address_has_been_spammed(self, address: ChecksumEvmAddress) -> bool:
+        """
+        Queries erc20 tranfers for the given address and if it has only transfer of spam assets
+        or ignored assets we return True. Stop at the first valid ERC20 transfer we find to exit
+        as early as possible. If any transfer had an unknown asset mark the address as not spammed
+        since we can't do more to classify it.
+        """
+        start_ts, end_ts = Timestamp(0), ts_now()
+        checked_tokens = set()
+        with self.database.conn.read_ctx() as cursor:
+            ignored_assets = self.database.get_ignored_asset_ids(cursor)
+
+        log.debug(f'Address detection: querying {self.evm_inquirer.chain_name} ERC20 Transfers for {address} -> {start_ts} - {end_ts}')  # noqa: E501
+        try:
+            for erc20_tx_hashes in self.evm_inquirer.etherscan.get_token_transaction_hashes(
+                account=address,
+                from_ts=start_ts,
+                to_ts=end_ts,
+            ):
+                for tx_hash in erc20_tx_hashes:
+                    raw_receipt_data = self.evm_inquirer.get_transaction_receipt(deserialize_evm_tx_hash(tx_hash))  # noqa: E501
```

Comment:
Hmm I just noticed that `get_token_transaction_hashes` returns a str. But this makes no sense. Can you please force it to return an evm transaction directly so that both here and in the one other place it's used we get a proper EvmTransaction?

What's more make sure to handle any deserialization errors (though if etherscan behaves okay there should be none)